server/middleware/auth.go

```diff
@@ -162,13 +166,18 @@ func authFunction(ctx context.Context, jwtValidators []*validator.Validator, con
 		return ctx, err
 	}
 
-	validatedToken, found := cache.Get(tkn)
-	log.Debug().Bool("validated token found", found).Interface("token", validatedToken)
-	if !found {
+	var validatedToken interface{}
+	if !BypassAuthCaches(ctx) {
```

Comment:
Can we move this logic in a separate function, then it would be easy to test whether a token is returned from the cache or not when the header is set by adding the token in the cache and then setting the header. 

```
validatedToken := getFromCache()
```

Reply:
added test for this. Thanks